### PR TITLE
fix(scannerv4): Move ScannerV4 client into `pkg`

### PR DESCRIPTION
**Pull request by dcaravel**
## Description

Moves the ScannerV4 client from
```
scanner/pkg/client
``` 
to 
```
pkg/scannerv4/client
```
so that the the new ScannerV4 scanner integration (which will be created at `pkg/scanners/scannerv4`) can use the client while making `roxvet` happy ([ScannerV4 Integration PR](https://github.com/stackrox/stackrox/pull/8869))

If the client is kept as is, `roxvet` complains: `pkg/scanners/scannerv4/scannerv4.go:17:2: invalid import "github.com/stackrox/rox/scanner/pkg/client": pkg cannot import from scanner/pkg/client; only allowed roots are [pkg generated image]`.

Other options to consider:

1. Change `roxvet` to allow the import from `scanner/pkg/client`
2. Move all scanner integrations from `pkg/scanners` to somewhere under `central/` and keep `scanner/pkg/client` as is
3. Something else?

## Checklist
- [x] Investigated and inspected CI test results
- [ ] ~Unit test and regression tests added~
- [ ] ~Evaluated and added CHANGELOG entry if required~
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Tested via unit tests and the [ScannerV4 Integration PR](https://github.com/stackrox/stackrox/pull/8869)

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by RTann — APPROVED** [review]
I think we can remove these lines from here too: https://github.com/stackrox/stackrox/blob/master/tools/roxvet/analyzers/validateimports/analyzer.go#L333

File: scanner/cmd/scannerctl/main.go [hunk 1]
```diff
@@ -14,10 +14,10 @@ import (
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/spf13/cobra"
 	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/pkg/scanners/scannerv4/client"
```
Comment (jvdm) [inline scanner/cmd/scannerctl/main.go:17]: I just realized that I forgot about `scannerctl`. It's quite ugly to have the scanner command line dependent on `rox/pkg/scanners`, which contains a lot of code and packages related to image integrations. Is it possible to move this to a different location under `rox/pkg/`? I think `pkg/scannerv4/client` would be a good option. Alternatively, `pkg/scanner/client` could also work, but having `pkg/scanners`, `pkg/scanner`, and `pkg/scan` is confusing, so `pkg/scannerv4` might be the best choice.

Comment (dcaravel) [reply scanner/cmd/scannerctl/main.go:17]: Alrighty, will put it under `pkg/scannerv4` 

Perhaps this is a style choice, but I believe its cleaner to have a single `pkg/scanners` directory to hold all 're-usable stuff for scanners'  - the `scannersv4/client` subdir doesn't necessarily mean it's associated with an integration.

**Pull request merged**